WLC/utils/gui.py

```diff
@@ -23,28 +20,31 @@ def init_button(self):
 
     # Execute code and display text field accordingly
     def execute_code(self, picture):
-        code_executor = CodeExecutor(self.docker_ip, DEFAULT_DOCKER_PORT)
-        image = Preprocessor().process(picture)
-        code, indents, poss_lines = image.get_code()
-        code = code.lower()
-        fixed_code = TrialCodeFixer(code, indents, poss_lines).fix()
-        value = str(code_executor.execute_code(fixed_code))
-        self.display_ocr(fixed_code)
-        self.display_output(value)
+        unfixed_code, fixed_code, result, error = CodeExecutor(self.docker_ip, DEFAULT_DOCKER_PORT).execute_code(picture)
+        self.display_ocr(unfixed_code, fixed_code)
+        self.display_output(result, error)
 
     # Set up text field for OCR output
-    def display_ocr(self, code):
+    def display_ocr(self, unfixed_code, fixed_code):
         text = tk.Text(self.master, height=14, width=70)
         text.pack(side=tk.LEFT, fill=tk.Y)
-        text.insert(tk.INSERT, "CODE READ BY OCR : \n")
-        text.insert(tk.END, code)
+        text.insert(tk.INSERT, "UNFIXED CODE READ BY OCR: \n")
+        text.insert(tk.END, unfixed_code)
+
+        text.insert(tk.INSERT, "\nCODE AFTER FIXING: \n")
```

Comment:
Do we want to stick both into same text field?

Reply:
Well that is up to you, if we plan user to edit the code and resubmit even in GUI then we need to split these (as user would be editing fixed code only)

Reply:
This is done now.